### PR TITLE
MLIR#545: Use upstream libmlir_rocm_runtime instead of librocm-runtime-wrappers

### PR DESCRIPTION
**Pull request by sjw36**
...for MLIR supported calls

- updated all mlir-rocm-runner tests

https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/545

### CODE REVIEW

**Comment by sjw36** [comment]
Using %linalg_test_lib_dir since it points to the external/llvm-project/llvm/lib dir but perhaps we should rename it..

**Comment by krzysz00** [comment]
Quick question: did you get the test generation files in `mlir/utils/`?

**Comment by krzysz00** [comment]
I take it there's a future PR where we migrate to the `gpu.*` ops that lower to the external calls?

**Comment by sjw36** [comment]
> I take it there's a future PR where we migrate to the `gpu.*` ops that lower to the external calls?

Exactly!

**Comment by sjw36** [comment]
> Quick question: did you get the test generation files in `mlir/utils/`?

Everything in there seems to be using our locally added routines..

**Comment by krzysz00** [comment]
@sjw36 What I'm asking about are the `.toml` files that are used to the e2e tests - those have a "suffix" parameter that includes the rocm-runner string. That needs to be changed to the new string to ensure we don't break things when someone uses the script to regenerate the tests.

**Comment by sjw36** [comment]
@krzysz00 please re-review/approve.

File: mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp [hunk 1]
```diff
@@ -276,14 +222,6 @@ extern "C" void mcpuMemset(float *allocated, float *aligned, int64_t offset,
...
 
-extern "C" StridedMemRefType<float, 1>
-mgpuMemAlloc(float *allocated, float *aligned, int64_t offset, int64_t size,
-             int64_t stride) {
-  float *gpuPtr;
-  hipMalloc((void **)&gpuPtr, size * sizeof(float));
-  return {gpuPtr, gpuPtr, offset, {size}, {stride}};
-}
-
 extern "C" void mgpuMemDealloc(float *allocated, float *aligned, int64_t offset,
```
Comment (krzysz00) [inline mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp:225]: I take it they don't have a free() upstream?

**Review by krzysz00 — APPROVED** [review]

**Comment by krzysz00** [comment]
Good cleanup, let's get this in!

File: mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp (hunk 1 above)
Comment (sjw36) [reply mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp:225]: Nope..

**Pull request merged**